db/python/layers/ourdna/dashboard.py

```diff
@@ -95,129 +101,231 @@ def get_collection_to_process_start_time(self, sample: Sample) -> int | None:
 
         return int(time_taken.total_seconds())
 
+    def fetch_key_from_meta(self, key: str, meta: str | dict[str, Any]) -> bool:
+        """
+        Fetches a key from the given meta, if it exists, and checks if it is not False or None
+        """
+        if isinstance(meta, str):
+            meta = json.loads(meta)
```

Comment:
Does this happen often? Feels performant dangerous to potentially always deserialise a JSON string